src/libponyc/pass/flatten.c

```diff
@@ -40,13 +40,26 @@ static void flatten_isect_element(ast_t* type, ast_t* elem)
   ast_free_unattached(elem);
 }
 
-static ast_result_t flatten_isect(ast_t* ast)
+static ast_result_t flatten_isect(typecheck_t* t, ast_t* ast)
 {
   // Flatten intersections without testing subtyping. This is to preserve any
   // type guarantees that an element in the intersection might make.
```

Comment:
for future reference, comments like this is expanded on or linked to more explanatory documentation would help new people learn the compiler. why does not testing subtyping when flattening intersections preserve type guarantees?